Create `Build retention` page. Update `Build export` page (#2062)
buildkite/docs: pages/pipelines/build_retention.md

# Build retention

> Each [Buildkite plan](https://buildkite.com/pricing) has a maximum build retention period. Once your builds reach the retention period, they will be removed from Buildkite.

Reviewer: Sorry, I missed commenting on this in the last round. Style nitpicks:

```suggestion
Each [Buildkite plan](https://buildkite.com/pricing) has a maximum build retention period. Once builds reach the retention period, their data is removed from Buildkite.
```

The following diagram shows the lifecycle of build data by plan.

<%= image "build-retention-flow-chart.png", alt: "Simplified flow chart of the build retention process" %>

## Retention periods

<table width="100%">
  <thead>
    <tr>
      <th>Plan</th>
      <th>Retention period</th>
      <th>Supports build exports</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Developer/Free plan</td>
      <td>90 days</td>
      <td>-</td>
    </tr>
    <tr>
    <tr>
      <td>Open source plan</td>
      <td>1 year</td>
      <td>-</td>
    </tr>
    <tr>
      <td>Non-profits & charities plan</td>
      <td>1 year</td>
      <td>-</td>
    </tr>
      <td>Team plan</td>
      <td>1 year</td>
      <td>-</td>
    </tr>
    <tr>
      <td>Business plan</td>
      <td>1 year</td>
      <td>-</td>
    </tr>
    <tr>
      <td>Enterprise plan</td>
      <td>1 year</td>
      <td>Yes</td>
    </tr>
  </tbody>
</table>

> ## Exporting build data

Reviewer: Add blank line after heading:

```suggestion
## Exporting build data

```
